### PR TITLE
[DEVOPS-819] asset-lock integration test and demo cluster support

### PR DESCRIPTION
**Pull request by disassembler**
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: wallet-new/integration/Functions.hs [hunk 1]
```diff
@@ -610,20 +610,6 @@ runAction wc action = do
...
-    -- Let's print it out to JSON
-    walletState <- get
-
-    log "=================================================================="
-    log . decodeUtf8 . encodePretty $ diff (toJSON previousWalletState) (toJSON walletState)
-    log "=================================================================="
-
-    -- increment successful actions
-    log "Success!"
-    successActions <>= [action]
```
Comment (parsonsmatt) [inline wallet-new/integration/Functions.hs]: Why is this removed?

File: wallet-new/integration/Functions.hs [hunk 2]
```diff
@@ -38,8 +38,8 @@ import           Cardano.Wallet.Client (ClientError (..), Response (..), Servant
 import           Pos.Core (getCoin, mkCoin, unsafeAddCoin, unsafeSubCoin)
 import qualified Pos.Wallet.Web.ClientTypes.Types as V0
 
-import           Error
-import           Types
+import           Error ( WalletTestError(..), showConstr )
+import           Types (WalletState, ActionProbabilities, WalletTestMode, wallets, getWeight, lastAction, successActions, transactions, addresses, walletsPass, accounts, nonGenesisWallets, actionsNum, Action (..))
```
Comment (parsonsmatt) [inline wallet-new/integration/Functions.hs]: For internal modules that we use heavily, adding explicit imports only adds noise IMO. This is also way over the 80 char column limit

File: default.nix [hunk 3]
```diff
@@ -89,6 +89,30 @@ let
       mkDerivation = args: super.mkDerivation (args // {
         enableLibraryProfiling = enableProfiling;
         enableExecutableProfiling = enableProfiling;
+        } // optionalAttrs (args ? src) {
```
Comment (cleverca22) [inline default.nix]: minor nit-pick, id have one less indent on this line

**Comment by erikd** [comment]
Pushing one more tiny fix to appease AppVeyor.
